apps/autoscheduler/CMakeLists.txt

```diff
@@ -0,0 +1,107 @@
+# weights
+file(GLOB WEIGHTS_DATA_FILES weights/*.data)
+set(WEIGHTS_CPP_FILES)
+
+foreach(wf ${WEIGHTS_DATA_FILES})
+  get_filename_component(base_name ${wf} NAME_WE)
+  set(wcpp ${base_name}.cpp)
+
+  add_custom_command(OUTPUT ${wcpp}
+                     DEPENDS ${wf} binary2cpp
+                     COMMAND binary2cpp weights_${base_name} < ${wf} > ${wcpp}
+                     COMMENT "${wf} -> ${wcpp}"
+                     VERBATIM)
+  list(APPEND WEIGHTS_CPP_FILES ${wcpp})
+endforeach()
+
+# =======================================================
+
+# train_cost_model
+add_executable(train_cost_model_process
+               ASLog.cpp
+               ASLog.h
+               train_cost_model.cpp
+               DefaultCostModel.cpp
+               CostModel.h
+               NetworkSize.h
+               ${WEIGHTS_CPP_FILES})
+set_target_properties(train_cost_model_process
+                      PROPERTIES OUTPUT_NAME train_cost_model)
+
+set(COST_MODEL_GENERATOR_FILES
+    cost_model_generator.cpp
+    cost_model_schedule.h
+    NetworkSize.h)
+halide_generator(cost_model.generator SRCS ${COST_MODEL_GENERATOR_FILES})
+halide_generator(train_cost_model.generator SRCS ${COST_MODEL_GENERATOR_FILES})
+halide_library_from_generator(cost_model
+                              FUNCTION_NAME
+                              cost_model
+                              GENERATOR
+                              cost_model.generator)
+halide_library_from_generator(train_cost_model
+                              FUNCTION_NAME
+                              train_cost_model
+                              GENERATOR
+                              train_cost_model.generator)
+target_link_libraries(train_cost_model_process
+                      PRIVATE cost_model train_cost_model)
+add_dependencies(train_cost_model_process cost_model train_cost_model)
+
+# =======================================================
+
+# libauto_schedule
+add_library(libauto_schedule
```

Comment:
In theory we should need to somehow specify the equivalent of USE_EXPORT_DYNAMIC here (ie, `-rdynamic` for linux, `-undefined dynamic_lookup` for osx)... except I'm not sure if such a thing exists for Windows. Does this actually link?